client/app/components/validator/validator.service.js

```diff
@@ -1,128 +1,123 @@
-angular.module("services").service("Validator", [
-    function () {
-        "use strict";
+/* eslint-disable class-methods-use-this, max-len */
+{
+    const IPV4_REGEX = /^(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])$/;
```

Comment:
Maybe `validator-js` dependency could help you?

Reply:
Actually has to depart from `validator.js` as this service is to validate data according to OVH rules, not according to the specifications.

Reply:
Why validate a IPv4 should be different in your case ? (excluding some IP for example ?)

If the API has custom rules to validate an IP. In this case, you should get the pattern to use directly by API, and don't manage it in the front-end.

What are the difference between an IP and an IP in this case ?